lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import userEvent from '@testing-library/user-event';
+
+import NxCombobox, { Props } from '../NxCombobox';
+
+describe('NxCombobox', function() {
+  const minimalProps: Props<string | number> = {
+        value: '',
+        onChange: () => {},
+        onSearch: () => {},
+        matches: []
+      },
+      quickRender = rtlRender(NxCombobox, minimalProps),
+      renderEl = rtlRenderElement(NxCombobox, minimalProps);
+
+  it('sets specified classNames and attributes on the top-level element', function() {
+    const component = renderEl({ className: 'foo', title: 'bar' });
+
+    expect(component).toHaveClass('foo');
+    expect(component).toHaveAttribute('title', 'bar');
+  });
+
+  it('adds the ref to the top-level element', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        domNode = render(<NxCombobox { ...minimalProps } ref={ref} />).container.firstElementChild!;
+
+    expect(ref.current).toBeDefined();
+    expect(ref.current).toBe(domNode);
+  });
+
+  it('renders an input child with the combobox role', function() {
+    const inputElement = quickRender().getByRole('combobox');
+
+    expect(inputElement.tagName).toBe('INPUT');
+    expect(inputElement).toBeInTheDocument();
+  });
+
+  it('sets the value as the value of the input', function() {
+    expect(quickRender({ value: 'foo' }).getByRole('combobox')).toHaveValue('foo');
+  });
+
+  it('sets aria-autocomplete on the input to list as default and to both if `autoComplete` prop is set to true',
+      function() {
+        const { getByRole, rerender } = quickRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox { ...minimalProps } autoComplete={true}/>);
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const { getByRole, rerender } = quickRender({
+              autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }]
+            }),
+            inputElement = getByRole('combobox');
+
+        await userEvent.type(inputElement, 'f');
+        rerender(
+          <NxCombobox { ...minimalProps }
```

Comment:
This test setup doesn't quite make sense to me, shouldn't the user's typed `value` ("f") be present in the re-rendered props if want a realistic situation?

Reply:
> This test setup doesn't quite make sense to me, shouldn't the user's typed value ("f") be present in the re-rendered props if want a realistic situation?

Actually it sounds like that likely won't work: https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection#related_objects

Reply:
When reviewing this, I found the original test set up does not update the value in jest dom, and rerender component will set the value to whatever the first option is in the matches which is not correct. I update the test and have the useState here to help update the value. 6dbc601488edec3acdf06ca1c289688de300a1a8.

Reply:
> When reviewing this, I found the original test set up does not update the value in jest dom, and rerender component will set the value to whatever the first option is in the matches which is not correct. I update the test and have the useState here to help update the value. https://github.com/sonatype/sonatype-react-shared-components/commit/6dbc601488edec3acdf06ca1c289688de300a1a8.

Is there a reason you can't just do a simple render with `autoComplete`, these `matches`, and `value: 'f'`? That should produce the result your looking for shouldn't it?

Reply:
With the simple render, I would have to use `user.type(inputElement, 'f')` to trigger the `inlineStyle` to set to `true` in order for it to work. Updated here 7047f0760fdf092735b418b118ce8b2643a1d17c, thanks.